src/interfaces/token.ts

```diff
@@ -123,15 +127,63 @@ export class TokenInterface<C extends ChainId> extends ServiceInterface<C> {
   }
 
   /**
-   * Approve vault to spend a token on zapIn
+   * Get allowance for a token either zapIn or direct deposit
    * @param vault
    * @param token
    * @param amount
    * @param account
    * @returns transaction
    */
-  async approve(
-    vault: Vault,
+  async allowance(
+    vault: { address: string; token: string },
+    token: Address,
+    account: Address
+  ): Promise<TokenAllowance> {
+    // If Ether is being sent, no need for approval
+    let allowance: TokenAllowance = {
+      owner: account,
+      spender: vault.address,
+      amount: MaxUint256.toString(),
+      token
+    };
+    debugger;
+    if (EthAddress === token) return allowance;
+
+    if (vault.token === token) {
+      const tokenContract = new Contract(token, TokenAbi, this.ctx.provider.read);
+      const addressToApprove =
+        (this.shouldUsePartnerService(vault.address) && this.yearn.services.partner?.address) || vault.address;
+      const allowanceAmount = await tokenContract.allowance(account, addressToApprove);
+
+      return {
+        ...allowance,
+        spender: addressToApprove,
+        amount: allowanceAmount.toString()
+      };
+    }
+
+    const zapProtocol = PickleJars.includes(vault.address) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
+
+    const zapInApprovalState = await this.yearn.services.zapper.zapInApprovalState(account, token, zapProtocol);
+
+    return {
+      owner: zapInApprovalState.ownerAddress,
+      spender: zapInApprovalState.spenderAddress,
+      amount: zapInApprovalState.allowance,
+      token
+    };
+  }
+
+  /**
+   * Approve vault to spend a token on zapIn or direct deposit
+   * @param vault
+   * @param token
+   * @param amount
+   * @param account
+   * @returns transaction
+   */
+  async approveDeposit(
+    vault: { address: string; token: string },
```

Comment:
Can we separate this two as different parameters, and not as an object?

Reply:
@xgambitox what do you think of moving towards named params instead of positional ones? Like this the order doesn't matter, and as they are both `string` it will be a little bit safer

```typescript
approveDeposit({ address: '0x', token: 'foo' }); // safer?

approveDeposit('0x', 'foo');
```

Reply:
Yeah, we use that approach in v3 frontend and would prefer it here too,  since changes to a function interface can affect parameter position and pass unnoticed. Maybe for SDK v2, since it might be a big breaking change for all interfaces.

Reply:
fixed on 4a954ae5d06e2e712fb3bc2e17b9c73c4a96a565